metabase/metabase
# feat(sdk): Add Dashboard loading event handlers

### Description

This adds dashboard load handlers to the SDK dashboard components - StaticDashboard, InteractiveDashboard:
* `onLoad?: (dashboard: Dashboard | null) => void;`
* `onLoadWithCards?: (dashboard: Dashboard | null) => void;`

Also, apart from props handlers, this adds global event handlers passed to `MetabaseProvider`


### How to verify

- Check this changes with https://github.com/metabase/shoppy/tree/testing-dashboard
- Go to https://localhost:3004/dash



### Checklist

- [x] Tests have been added/updated to cover changes in this PR - **I will add tests for this in upcomming days.**
